packages/@azure/servicebus/data-plane/lib/util/utils.ts

```diff
@@ -7,6 +7,13 @@ import { generate_uuid, string_to_uuid } from "rhea-promise";
 import { isBuffer } from "util";
 import { ConnectionContext } from "../connectionContext";
 
+// This is the only dependency we have on DOM types, so rather than require
+// the DOM lib we can just shim this in.
+interface Navigator {
+  hardwareConcurrency: number;
+}
+declare var navigator: Navigator;
+
```

Comment:
How does this affect when we include browser support?

Reply:
It doesn't really affect anything, this is a type-only thing. With browser support, there are additional DOM library dependencies (e.g. on WebSocket). I'll think about the best way to handle that over in amqp-common. I'm thinking we'll want to shim WebSocket as well.